internal/install/discovery/regex_process_filterer.go

```diff
@@ -23,45 +23,68 @@ func NewRegexProcessFilterer(r recipes.RecipeFetcher) *RegexProcessFilterer {
 	return &f
 }
 
-func (f *RegexProcessFilterer) filter(ctx context.Context, processes []types.GenericProcess) ([]types.GenericProcess, error) {
+func (f *RegexProcessFilterer) filter(ctx context.Context, processes []types.GenericProcess) ([]types.ProcessInfoWrap, error) {
+	processesInfo := getProcessesInfo(processes)
+	log.Debugf("Filtering recipes with %d processes...", len(processesInfo))
+	for _, p := range processesInfo {
+		log.Debugf("Match using processInfo: %s", p.Info)
+	}
+
 	recipes, err := f.recipeFetcher.FetchRecipes(ctx)
 	if err != nil {
 		return nil, fmt.Errorf("could not retrieve process filter criteria: %s", err)
 	}
+	for _, r := range recipes {
+		log.Debugf("Match using recipe DisplayName: %s RecipeProcessMatch: %s", r.DisplayName, r.ProcessMatch)
+	}
 
-	matches := []types.GenericProcess{}
-	for _, p := range processes {
+	matches := []types.ProcessInfoWrap{}
+	for _, p := range processesInfo {
 		isMatch := false
 		for _, r := range recipes {
-			isMatch = isMatch || match(r, p)
+			isMatch = isMatch || match(r, &p)
 		}
 
 		if isMatch {
 			matches = append(matches, p)
 		}
 	}
 
+	log.Debugf("Filtering recipes with processes done, found %d matches.", len(matches))
 	return matches, nil
 }
 
-func match(r types.Recipe, process types.GenericProcess) bool {
+func match(r types.Recipe, processInfo *types.ProcessInfoWrap) bool {
 	for _, pattern := range r.ProcessMatch {
-		name, err := process.Name()
+		matched, err := regexp.Match(pattern, []byte(processInfo.Info))
 		if err != nil {
-			log.Debugf("could not retrieve process name for PID %d", process.PID())
-			continue
-		}
-
-		matched, err := regexp.Match(pattern, []byte(name))
-		if err != nil {
-			log.Debugf("could not execute pattern %s against process invocation %s", pattern, name)
+			log.Debugf("could not execute pattern %s against process invocation %s", pattern, processInfo.Info)
 			continue
 		}
 
 		if matched {
+			processInfo.MatchingPattern = pattern
+			log.Debugf("Process matching pattern %s with %s for recipe %s.", pattern, processInfo.Info, r.DisplayName)
 			return matched
 		}
 	}
 
 	return false
 }
+
+func getProcessesInfo(processes []types.GenericProcess) []types.ProcessInfoWrap {
```

Comment:
Consider renaming to `getRunningProcesses` and returning a slice of `GenericProcess` from the method here?  We can resolve the `CmdLine()` values when we perform the matching, we don't have to do it here.